strax/config.py

```diff
@@ -203,6 +218,224 @@ def validate(self, config,
             config[self.name] = self.get_default(run_id, run_defaults)
 
 
+#Backward compatibility
+@export
+class Config(Option):
+    def __init__(self, **kwargs):
+        if 'name' not in kwargs:
+            kwargs['name'] = ''
+        super().__init__(**kwargs)
+
+    def __set_name__(self, owner, name):
+        self.name = name
+        takes_config = {name: self}
+        if (hasattr(owner, 'takes_config')
+                and len(owner.takes_config)):
+            # Already have some options set, e.g. because of subclassing
+            # where both child and parent have a takes_config decorator
+            
+            if name in owner.takes_config:
+                raise RuntimeError(
+                    f"Attempt to specify option {name} twice")
+            owner.takes_config = immutabledict({
+                **owner.takes_config, **takes_config})
+        else:
+            owner.takes_config = immutabledict(takes_config)
+
+    def __get__(self, obj, objtype=None):
+        return self.fetch(obj)
+
+    def __set__(self, obj, value):
+        obj.config[self.name] = value
+
+    def fetch(self, plugin):
+        ''' This function is called when the attribute is being 
+        accessed. Should be overriden by subclasses to customize behavior.
+        '''
+        if hasattr(plugin, 'config') and self.name in plugin.config:
+            return plugin.config[self.name]
+        raise AttributeError('Plugin has not been configured.')
+
+@export
+class LookupConfig(Config):
+    mapping: ty.Mapping
+    keys = ty.Iterable
+
+    def __init__(self, mapping: ty.Mapping, keys=('name', 'value'), **kwargs):
+        super().__init__(**kwargs)
+        self.mapping = mapping
+        if not isinstance(keys, ty.Iterable):
+            keys = (keys,)
+        self.keys = keys
+        
+    def fetch(self, plugin):
+        key = []
+        for k in self.keys:
+            if k=='name':
+                v = self.name
+            elif k=='value':
+                v = plugin.config[self.name]
+            elif isinstance(k, str) and hasattr(plugin, k):
+                v = getattr(plugin, k)
+            else:
+                v = k
+            key.append(v)
+        if len(key)==1:
+            key = key[0]
+        else:
+            key = tuple(key)
+        return self.mapping[key]
+@export
+class RemoteConfig(Config):
```

Comment:
I can imagine some slight issues concerning reproducibility with this class, if a value can come from any storage, and it's up to the user to make sure that either the storage's all provide unique names or that they always return the same. I know that to some extend this can also easily be argued for the other Options, but at least there you know more or less the hopefully unique path to the requested value. We don't track which storage is providing the requested information, this might both be a feature or a bug, depending fully on the setup one chooses for 😉.